backend/src/main/scala/cromwell/backend/WdlStandardLibraryImpl.scala

```diff
@@ -0,0 +1,180 @@
+
+    def validateArguments(values: (Try[WdlValue], Try[WdlValue], Try[WdlValue])) = values match {
+      case (Success(strValue), Success(WdlString(pattern)), Success(replaceValue))
+        if WdlStringType.isCoerceableFrom(strValue.wdlType) &&
+          WdlStringType.isCoerceableFrom(replaceValue.wdlType) =>
+        Success((strValue.valueString, pattern, replaceValue.valueString))
+      case _ => Failure(new IllegalArgumentException(s"Invalid parameters for engine function sub: $values."))
+    }
+
+    for {
+      args <- extractArguments
+      (str, pattern, replace) <- validateArguments(args)
+    } yield WdlString(pattern.r.replaceAllIn(str, replace))
+  }
+
+  /**
+    * Try to read a boolean from the file referenced by the specified `WdlValue`.
+    */
+  override def read_boolean(params: Seq[Try[WdlValue]]): Try[WdlBoolean] =
+    read_string(params) map { s => WdlBoolean(java.lang.Boolean.parseBoolean(s.value.trim.toLowerCase)) }
+
+  private def writeToTsv(params: Seq[Try[WdlValue]], wdlClass: Class[_ <: WdlValue with TsvSerializable]) = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      downcast <- Try(wdlClass.cast(singleArgument))
+      tsvSerialized <- downcast.tsvSerialize
+      file <- writeContent(wdlClass.getSimpleName.toLowerCase, tsvSerialized)
+    } yield file
+  }
+
+  override def size(params: Seq[Try[WdlValue]]): Try[WdlFloat] = {
+    def toUnit(wdlValue: Try[WdlValue]) = wdlValue flatMap { unit => Try(MemoryUnit.fromSuffix(unit.valueString)) }
+
+    def fileSize(wdlValue: Try[WdlValue], convertTo: Try[MemoryUnit] = Success(MemoryUnit.Bytes)) = {
+      for {
+        value <- wdlValue
+        unit <- convertTo
+      } yield MemorySize(adjustFilePath(value.valueString).toAbsolutePath.size.toDouble, MemoryUnit.Bytes).to(unit).amount
+    }
+
+    params match {
+      case oneArg if params.length == 1 => fileSize(params.head) map WdlFloat.apply
+      case twoArgs if params.length == 2 => fileSize(params.head, toUnit(params(1))) map WdlFloat.apply
+      case _ => Failure(new UnsupportedOperationException(s"Expected one or two parameters but got ${params.length} instead."))
+    }
+  }
+
+  override def write_lines(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv(params, classOf[WdlArray])
+  override def write_map(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv(params, classOf[WdlMap])
+  override def write_object(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv(params, classOf[WdlObject])
+  override def write_objects(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv(params, classOf[WdlArray])
+  override def write_tsv(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv(params, classOf[WdlArray])
+  override def writeTempFile(path: String, prefix: String, suffix: String, content: String): String = {
+    // This may be called multiple times with the same inputs.  Calling this twice with the same
+    // parameters should yield the same return value.
```

Comment:
It would a good idea to wait for @Horneth PR merge and after that I will rebase and merge my PR (when we solve the issue of sharing data and code duplication.